File: nimbus/Components/TableCellSynthesis.m
//
//  TableCellSynthesis.m
//  nimbus
//
//  Created by Pericles Maravelakis on 02/07/2018.
//	periclesm@cloudfields.net
//	Licensed under Creative Commons Attribution-ShareAlike 4.0 International (CC BY-SA 4.0)
//	https://creativecommons.org/licenses/by-sa/4.0/
//

#import "TableCellSynthesis.h"

@implementation TableCellSynthesis

+ (cloudCell*)CloudCell:(UITableView*)tableView datasource:(NSArray*)datasource index:(NSIndexPath*)indexPath
{
	cloudCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cloudCell" forIndexPath:indexPath];
	CloudList *cl = datasource[indexPath.row];

	cell.cloudInitials.text = cl.initials;
	cell.cloudName.text = cl.name;
	cell.cloudDetail.text = [DataLogic GetCloudDetails:cl.detail shortText:YES];

	return cell;
}

@end

File: nimbus/View/Components/TableCells/TableCellSynthesis.swift
//
//  TableCellSynthesis.swift
//  nimbus
//
//  Created by Pericles Maravelakis on 1/2/20.
//  Copyright © 2020 Cloudfields. All rights reserved.
//

import UIKit

class TableCellSynthesis: NSObject {
    
    class func cloudCell(for tableView: UITableView, datasource: Array<Any>, index: IndexPath) -> cloudCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cloudCell", for: index) as! cloudCell
        let cl = datasource[index.row] as! Cloud
        
        cell.cloudInitials?.text = cl.initials
        cell.cloudName?.text = cl.name
		cell.cloudDetail?.text = cl.excerpt
        
        return cell
    }
}
